src/dnnl/dnnl_graph_compiler.cpp

```diff
@@ -43,54 +41,110 @@
 
 // dnnl_graph_compiler.h interface implementation.
 
+using namespace mlir;
+
+extern "C" {
+extern int gc_runtime_keep_alive;
+}
+
 struct dnnl_graph_compiler_executable {
-  // TODO: Implement
+  dnnl_graph_compiler_executable(MLIRContext &context,
+                                 const std::string_view &json)
+      : inputIds(), outputIds(), jmod() {
+    // keeps GCCPURuntime linked
+    gc_runtime_keep_alive++;
+    auto mod = JsonParser::parse(context, json, inputIds, outputIds, strides);
+    auto jmod = gc::JitModule::create(mod);
+
+    if (!static_cast<bool>(jmod)) {
+      auto err = jmod.takeError();
+      llvm::errs() << err;
```

Comment:
Shall we dump the error to `llvm::raw_string_ostream` and pass the error message to the `std::runtime_error` below?

Reply:
Done.